frontend/src/app/App.tsx

```diff
@@ -10,26 +10,32 @@ import ColorReference, { MiscReference, SpaceReference } from "./components/refe
 import { ModalComponent as Modal } from "./components/modal/modal";
 import { useAppDispatch } from "./hooks/hooks";
 import { ZoneAtAGlance } from "./components/containers/zone-at-a-glance/zone-at-a-glance";
-import { fetchAllCodeTables } from "./store/reducers/code-table";
+import { fetchAllCodeTables, fetchComplaintCodeTables } from "./store/reducers/code-table";
 import { getOfficers } from "./store/reducers/officer";
 import { PageLoader } from "./components/common/page-loader";
 import { ComplaintsWrapper } from "./components/containers/complaints/complaints";
 import COMPLAINT_TYPES from "./types/app/complaint-types";
-import { getCodeTableVersion, getConfigurations, getOfficerDefaultZone } from "./store/reducers/app";
+import { getCodeTableVersion, getConfigurations, getFeatureFlag, getOfficerDefaultZone } from "./store/reducers/app";
 import { CreateComplaint } from "./components/containers/complaints/details/complaint-details-create";
 import { UserManagement } from "./components/containers/admin/user-management";
 import GenericErrorBoundary from "./components/error-handling/generic-error-boundary";
 import { VerifyAccess } from "./components/containers/pages/verify-access";
+import { FeatureManagement } from "./components/containers/admin/feature-management";
 
 const App: FC = () => {
   const dispatch = useAppDispatch();
 
   useEffect(() => {
     dispatch(getOfficerDefaultZone());
-    dispatch(fetchAllCodeTables());
+    if (Roles.CEEB) {
```

Comment:
This is just a thought, instead of having separate thunks for different roles/agencies it would be better to have one thunk and have it return the correct code-tables. This way it doesn't matter what your role or agency is, the code-table controller will return whats needed

Reply:
@marqueone-ps I'm not quite understand what you mean. You suggest having another thunk function with an agency as a parameter?